src/vs/editor/contrib/stickyScroll/browser/stickyScrollWidget.ts

```diff
@@ -109,7 +110,8 @@ export class StickyScrollWidget extends Disposable implements IOverlayWidget {
 		return this._lineNumbers;
 	}
 
-	setState(state: StickyScrollWidgetState | undefined): void {
+	setState(state: StickyScrollWidgetState | undefined, forceRebuild: boolean = false): void {
+		this._previousStickyLines = [...this._stickyLines];
```

Comment:
By storing it in a field, this array will be referenced also after `setState``, increasing the referenced memory. So I would only do that if needed. Here it's probably enough to pass it as an argument to `_renderRootNode`.
Also you create a copy of the array. That's not needed as `_clearStickyWidget` will just create a new array.